pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions_test.go

```diff
@@ -220,11 +243,186 @@ func TestISCSIDiskConflicts(t *testing.T) {
+			enableReadWriteOncePod: true,
+			wantStatus:             framework.NewStatus(framework.Success),
+		},
+		{
+			name:                   "failed to get PVC",
+			pod:                    podWithReadWriteOncePodPVC,
+			existingPods:           []*v1.Pod{},
+			existingNodes:          []*v1.Node{},
+			existingPVCs:           []*v1.PersistentVolumeClaim{},
+			enableReadWriteOncePod: true,
+			wantStatus:             framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonPVCNotFound),
+		},
+		{
+			name:                   "no access mode conflict",
+			pod:                    podWithReadWriteOncePodPVC,
+			existingPods:           []*v1.Pod{podWithReadWriteManyPVC},
+			existingNodes:          []*v1.Node{node},
+			existingPVCs:           []*v1.PersistentVolumeClaim{readWriteOncePodPVC, readWriteManyPVC},
+			enableReadWriteOncePod: true,
+			wantStatus:             framework.NewStatus(framework.Success),
+		},
+		{
+			name:                   "access mode conflict",
+			pod:                    podWithReadWriteOncePodPVC,
+			existingPods:           []*v1.Pod{podWithReadWriteOncePodPVC, podWithReadWriteManyPVC},
+			existingNodes:          []*v1.Node{node},
+			existingPVCs:           []*v1.PersistentVolumeClaim{readWriteOncePodPVC, readWriteManyPVC},
+			enableReadWriteOncePod: true,
+			wantStatus:             framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonReadWriteOncePodConflict),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			p, err := newPluginWithListers(test.existingPods, test.existingNodes, test.existingPVCs, test.enableReadWriteOncePod)
+			if err != nil {
+				t.Error(err)
+			}
+
+			gotStatus := p.(framework.PreFilterPlugin).PreFilter(context.Background(), nil, test.pod)
+			if !reflect.DeepEqual(gotStatus, test.wantStatus) {
+				t.Errorf("status does not match: %+v, want: %+v", gotStatus, test.wantStatus)
+			}
+		})
+	}
+}
+
+func newPlugin() (framework.Plugin, error) {
+	return newPluginWithListers([]*v1.Pod{}, []*v1.Node{}, []*v1.PersistentVolumeClaim{}, true)
+}
+
+func newPluginWithListers(pods []*v1.Pod, nodes []*v1.Node, pvcs []*v1.PersistentVolumeClaim, enableReadWriteOncePod bool) (framework.Plugin, error) {
+	ctx := context.Background()
+	informerFactory, err := populateAndStartInformers(ctx, pvcs)
+	if err != nil {
+		return nil, fmt.Errorf("error creating informer factory: %+v", err)
+	}
+
+	frameworkHandle, err := frameworkruntime.NewFramework(nil, nil,
+		frameworkruntime.WithInformerFactory(informerFactory),
+		frameworkruntime.WithSnapshotSharedLister(cache.NewSnapshot(pods, nodes)))
+	if err != nil {
+		return nil, fmt.Errorf("error creating new framework handle: %+v", err)
+	}
+	fts := feature.Features{EnableReadWriteOncePod: enableReadWriteOncePod}
+	return New(nil, frameworkHandle, fts)
+}
+
+func populateAndStartInformers(ctx context.Context, pvcs []*v1.PersistentVolumeClaim) (informers.SharedInformerFactory, error) {
```

Comment:
can you use [SetupPluginWithInformers](https://github.com/kubernetes/kubernetes/blob/41c3f5b58db08b2a1d17a2ebecd6cc6d3cb89086/pkg/scheduler/framework/plugins/testing/testing.go#L36) instead?